#! /bin/sh /usr/share/dpatch/dpatch-run
## server-clientid-crash.dpatch by Christoh Biedl <debian.packages.hhqj@manchmal.in-ulm.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Server assert involving client IDs (CVE-2009-1892)

@DPATCH@
diff -urNad git~/server/dhcp.c git/server/dhcp.c
--- git~/server/dhcp.c	2009-07-12 22:03:17.000000000 +0200
+++ git/server/dhcp.c	2009-07-12 22:04:42.000000000 +0200
@@ -1747,6 +1747,8 @@
 				host_reference (&host, h, MDL);
 		}
 		if (!host) {
+			if (hp)
+				host_dereference (&hp, MDL);
 			find_hosts_by_haddr (&hp,
 					     packet -> raw -> htype,
 					     packet -> raw -> chaddr,
